opv_tasks/task/exportviewertask.py

```diff
@@ -0,0 +1,179 @@
+from opv_tasks.task import Task, TaskStatusCode
+from opv_api_client import ressources, Filter
+from path import Path
+import tempfile
+import json
+
+class ExportviewerTask(Task):
+    """
+    Export virtual tour information into a static dir
+    Input format :
+        opv-task exportviewer '{"id_virtualtour": ID_VIRTUALTOUR, "id_malette": ID_MALETTE, "dir": DIR_TO_EXPORT }'
+    dir argument is optionnal
+    """
+    TASK_NAME = "exportviewer"
+    requiredArgsKeys = ["id_virtualtour", "id_malette"]
+
+    def createDir(self):
+        self.path.mkdir_p()
+        self.db = self.path / "db"
+        self.db.mkdir_p()
+        self.dm = self.path / "dm"
+        self.dm.mkdir_p()
+        self.virtualtour_dir = self.db / "virtualtour"
+        self.virtualtour_dir.mkdir_p()
+        self.panorama_dir = self.db / "panorama"
+        self.panorama_dir.mkdir_p()
+        self.logger.info("Working directory : {}".format(self.path))
+
+    def buildJSON(self):
+        self.panorama_to_add = []
+        self.virtualtour_dict = {
+            "id_virtualtour": self.virtualtour.id_virtualtour,
+            "id_malette": self.virtualtour.id_malette,
+            "title": self.virtualtour.title,
+            "description": self.virtualtour.decription,
+            "virtualtour_paths": []
+        }
+        for virtualtour_path in self.virtualtour_paths:
+            path_details = self._client_requestor.make(ressources.PathDetails, virtualtour_path.path_details.id_path_details, virtualtour_path.path_details.id_malette)
+            virtualtour_path_dict = {
+                "id_virtualtour_path": virtualtour_path.id_virtualtour_path,
+                "id_malette": virtualtour_path.id_malette,
+                "id_virtualtour": self.virtualtour.id_virtualtour,
+                "id_virutaltour_malette": self.virtualtour.id_malette,
+                "id_path_details": path_details.id_path_details,
+                "id_path_details_malette": path_details.id_malette,
+                "path_details": {
+                    "id_path_details": path_details.id_path_details,
+                    "id_malette": path_details.id_malette,
+                    "description": path_details.decription,
+                    "name": path_details.name,
+                    "id_campaign": path_details.campaign.id_campaign,
+                    "id_malette": path_details.campaign.id_malette,
```

Comment:
It should be path_details.id_malette
```suggestion
                    "id_malette": path_details.id_malette,
```

Reply:
We should change key to id_campaign_malette cause id_malette is set before